### PR TITLE
Generate source maps for processed `.js.rst` files.

### PR DESCRIPTION
**Pull request by daithiocrualaoich**
Browser Debugger tab with source map

### CODE REVIEW

File: examples/webpack-literacy/src/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+<html>
+<head>
+  <meta charset="UTF-8">
+  <title>Literacy Webpack Loader Example</title>
+</head>
+
+<body>
+    <h1>Hello, Doggies!</h1>
+
+    <script src='bundle.js'></script>
```
Comment (passy) [inline examples/webpack-literacy/src/index.html]: Nit: double-quotes

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "literacy",
   "description": "Literate programming in JavaScript using reStructuredText.",
-  "version": "0.2.5",
+  "version": "0.3.0",
```
Comment (passy) [inline package.json]: Most projects prefer version bumps to be atomic, stand-alone commits.

File: src/js/index.js.rst [hunk 3]
```diff
@@ -228,6 +230,78 @@ Otherwise, the line is regular text with optional indentation.
     const parser = peg.generate(rules.join(''));
 
 
+Source Maps
+-----------
+Calculate `source maps`_ for Literacy output. The ``.js.rst`` file format is
+impractical without access to debugger tools or stacktrace line number
+translation.
+
+.. _source map: https://www.html5rocks.com/en/tutorials/developertools/sourcemaps/
```
Comment (passy) [inline src/js/index.js.rst]: `maps`/`map` anchor doesn't seem to match.

**Review by passy — APPROVED** [review]
This was a joy to review!

File: package.json (hunk 2 above)
Comment (daithiocrualaoich) [reply package.json]: Fixed, thanks.

File: src/js/index.js.rst (hunk 3 above)
Comment (daithiocrualaoich) [reply src/js/index.js.rst]: Done.

File: src/js/index.js.rst [hunk 4]
```diff
@@ -304,19 +379,41 @@ that into the output. Note that the directive lines themselves are not given a
...
+      };
+
+If the input filename was provided then generate a source map. Without an input
+filename we cannot generate a valid source map. The source map created is line
+based, linking line numbers in the output to corresponding JavaScript block line
+numbers in the input.
+
+.. code-block:: javascript
+
+      if (typeof inputFilename !== 'undefined') {
```
[the remark this answers is not in the document]
Comment (daithiocrualaoich) [reply src/js/index.js.rst]: Yes, Your Holiness, forgive me.

Seriously, this area is a mess and it's been bothering me for a while. The problem is that the original implementation of the command line tool held too close to the Babel tool and provided too many features. Output options to stdout, file, and directory would be fine if the realization of it was tighter.

My thinking is to kill the stdout feature for starters.

For the particular sin at this line, rename the function `fromString` to `fromFileContent` and require the filename parameter. I like the idea of producing sourcemaps by default unless the caller explicitly suppresses them.

File: AUTHORS [hunk 5]
```diff
@@ -1 +1,2 @@
 Daithi O Crualaoich <daithi.ocrualaoich@gmail.com>
+Pascal Hartig <i@passy.me>
```
Comment (passy) [inline AUTHORS:2]: <3

**Comment by daithiocrualaoich** [comment]
Created #21 to handle the untidiness that has built up in `cli.js.rst`.

**Pull request merged**